drivers/serial/uart_mcux.c

```diff
@@ -410,11 +412,13 @@ static const struct uart_mcux_config uart_mcux_##n##_config = {		\
 									\
 		irq_enable(DT_INST_IRQ_BY_NAME(n, status, irq));	\
 									\
-		IRQ_CONNECT(DT_INST_IRQ_BY_NAME(n, error, irq),		\
-			    DT_INST_IRQ_BY_NAME(n, error, priority),	\
-			    uart_mcux_isr, DEVICE_DT_INST_GET(n), 0);	\
-									\
-		irq_enable(DT_INST_IRQ_BY_NAME(n, error, irq));		\
+		COND_CODE_0(DT_INST_PROP(n, no_error_irq), \
+		(IRQ_CONNECT(DT_INST_IRQ_BY_NAME(n, error, irq),	\
```

Comment:
I don't think this is needed, see other drivers like nxp enet for how to connect arbitrary number of IRQ

Reply:
Updated.